packaging/language/maven_artifact.py

```diff
@@ -257,7 +257,7 @@ def _request(self, url, failmsg, f):
         self.module.params['http_agent'] = self.module.params.get('user_agent', None)
 
         response, info = fetch_url(self.module, url_to_use, timeout=req_timeout)
-        if info['status'] != 200:
+        if info['status'] != 200 or info['msg'][:2] != 'OK':
```

Comment:
It might seem awkward, but these are actually the two cases where we want it to fail. It works as intended here... 

If the info['status'] != 200, it shouldn't automatically short circuit the logic and fail - it should perform a fallback check because maybe info['msg'] says OK.